### PR TITLE
MethodAccessor can manage args for the invoke method.

### PR DESCRIPTION
**Pull request by wru-talend**
**What is the problem this Pull Request is trying to solve?**
 
**What is the chosen solution to this problem?**
 
**Link to the JIRA issue**

 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-tql/daikon-tql-core/src/main/java/org/talend/tql/parser/Tql.java [hunk 1]
```diff
@@ -25,7 +26,7 @@ private Tql() {
      * @return A {@link TqlElement AST} of the query.
      */
     public static Expression parse(String query) {
-        ANTLRInputStream input = new ANTLRInputStream(query);
+        CharStream input = CharStreams.fromString(query);
```
Comment (fhuaulme) [inline daikon-tql/daikon-tql-core/src/main/java/org/talend/tql/parser/Tql.java]: Good idea 👍

File: daikon-tql/daikon-tql-bean/src/test/java/org/talend/tql/bean/BeanPredicateVisitorTest.java [hunk 2]
```diff
@@ -434,6 +436,42 @@ public void testParseFieldNotCompliesPatternWord() {
         assertFalse(predicate.test(bean));
     }
 
+    @Test
+    public void testMyMapWithAValidUnaryKey() {
```
Comment (fhuaulme) [inline daikon-tql/daikon-tql-bean/src/test/java/org/talend/tql/bean/BeanPredicateVisitorTest.java]: Can you add a test that doesn't match to be sure of implementation? (something like `attributes.version = '2.0'`)

File: daikon-tql/daikon-tql-bean/src/main/java/org/talend/tql/bean/MapMethodAccessor.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package org.talend.tql.bean;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
```
Comment (fhuaulme) [inline daikon-tql/daikon-tql-bean/src/main/java/org/talend/tql/bean/MapMethodAccessor.java]: Can you add some javadoc to explain what this implementation does?

File: daikon-tql/daikon-tql-bean/src/test/java/org/talend/tql/bean/BeanPredicateVisitorTest.java (hunk 2 above)
Comment (wru-talend) [reply daikon-tql/daikon-tql-bean/src/test/java/org/talend/tql/bean/BeanPredicateVisitorTest.java]: Done in https://github.com/Talend/daikon/pull/581/commits/e5be45260492109ebe5622417bf94bf1ae3b8f2e

File: daikon-tql/daikon-tql-bean/src/main/java/org/talend/tql/bean/MapMethodAccessor.java (hunk 3 above)
Comment (wru-talend) [reply daikon-tql/daikon-tql-bean/src/main/java/org/talend/tql/bean/MapMethodAccessor.java]: Done in https://github.com/Talend/daikon/pull/581/commits/e5be45260492109ebe5622417bf94bf1ae3b8f2e

**Pull request merged**